servicetalk-client-api/src/main/java/io/servicetalk/client/api/LoadBalancerFactory.java

```diff
@@ -35,8 +35,36 @@
      * and hence will call {@link ConnectionFactory#closeAsync()} when {@link LoadBalancer#closeAsync()} is called.
      * @param <T> Type of connections created by the passed {@link ConnectionFactory}.
      * @return a new {@link LoadBalancer}.
+     * @deprecated In the future only {@link #newLoadBalancer(String, Publisher, ConnectionFactory)} will remain,
+     * please use that method instead.
```

Comment:
Sad that we can not add a `default` impl for the deprecated method because it's a functional interface. Looks like we will need to swap `default` in 0.42 and will be able to remove this method only in 0.43. Do we need to provide any recommendations how users can migrate (what to do with the old method)?

Reply:
It seems because the functional interface's method is generic on its' own, the users can't use lambdas with it. Only method references. They should get compiler warnings and will be able to change their signatures to accept the String parameter and use the reference to the default method. WDYT?

Reply:
No, they won't be able to use lambda for a new method while it still has `default` impl. No worries, we will eventually clean it up, it will just take one more release to do so.